### PR TITLE
Set WinSDK version to use current ENV WindowsSDKVersion value when using MSVC

### PR DESCRIPTION
**Pull request by firodj**
I am using Cmake 3.14 with more than one windows sdk availables, eg.: 8.1.x, 10.0.x, ...
The generated solution use default to 8.1 (not set), which failed during build. It should be set to 10.x  regarding to WindowsSDKVersion ENV variable eg. "10.0.17763.0".

Resource:
* https://stackoverflow.com/questions/45692367/how-to-set-msvc-target-platform-version-with-cmake
* https://gitlab.kitware.com/cmake/cmake/issues/17992

### CODE REVIEW

**Comment by derekbruening** [comment]
Hmm, I would have expected the LIB, INCLUDE, and LIBDIR variables to be what counts and what point at the proper SDK, and that CMake doesn't go do anything on its own wrt the SDK version-- but maybe that's just the Ninja or other cmdline generators and the VS generator is different.

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,6 +59,11 @@ include(make/policies.cmake NO_POLICY_SCOPE)
 # this must be done prior to the project() command and the var
 # must be set in the cache.
 if ("${CMAKE_GENERATOR}" MATCHES "Visual Studio")
+  set(CMAKE_SYSTEM_VERSION_TEMP  $ENV{WindowsSDKVersion})
+  string(REGEX REPLACE "([0-9.]+).*" "\\1" CMAKE_SYSTEM_VERSION_TEMP "${CMAKE_SYSTEM_VERSION_TEMP}" )
+  if(CMAKE_SYSTEM_VERSION_TEMP)
```
Comment (derekbruening) [inline CMakeLists.txt]: style: As you can see this does not match all the other `if`'s as it has no space before the `(`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +59,11 @@ include(make/policies.cmake NO_POLICY_SCOPE)
 # this must be done prior to the project() command and the var
 # must be set in the cache.
 if ("${CMAKE_GENERATOR}" MATCHES "Visual Studio")
+  set(CMAKE_SYSTEM_VERSION_TEMP  $ENV{WindowsSDKVersion})
```
Comment (derekbruening) [inline CMakeLists.txt]: suggest: A comment explaining that this is a workaround until CMake issue 17992 is fixed would be helpful.

**Comment by AssadHashmi** [comment]
run arm tests

**Comment by AssadHashmi** [comment]
Added @firodj to the AArch64 precommit tests access list.

**Comment by derekbruening** [comment]
Appveyor failure is #3320 type_is_instr.

**Pull request merged**